[RIVER-1701] Remove support for legacy (unmigrated) stream schema

clemire opened the pull request:
#1701

clemire commented on core/node/storage/pg_stream_store.go:
```diff
@@ -376,7 +376,7 @@ func (s *PostgresStreamStore) createStreamArchiveStorageTx(
 			CREATE TABLE {{miniblocks}} PARTITION OF miniblocks FOR VALUES IN ($1);`,
 			streamId,
-			false,
+			s.config.MigrateStreamCreation,
```
Bugfix: the archiver was always using the legacy schema.

sergekh2 commented on core/node/testutils/dbtestutils/db.go:
```diff
@@ -76,14 +76,15 @@ func ConfigureDB(ctx context.Context) (*config.DatabaseConfig, string, func(), e
+			Extra:                 "?sslmode=disable&pool_max_conns=1000",
+			StartupDelay:          2 * time.Millisecond,
+			NumPartitions:         4,
+			MigrateStreamCreation: true,
```
after this change it seems this setting is no longer required?

clemire replied:
Yes, I was going to do in a follow-up PR but went ahead and put in here so the code would not be in a confusing state.

clemire commented:
@sergekh2 I updated this PR according to your comment to remove the migration setting, and i also updated the migration tool, but I did not test the migration tool since we don't really need it right now.  I saw some flakes locally but after making changes they disappeared - if 5 consecutive CI tests pass I will consider this solid.

clemire commented:
> @sergekh2 I updated this PR according to your comment to remove the migration setting, and i also updated the migration tool, but I did not test the migration tool since we don't really need it right now. I saw some flakes locally but after making changes they disappeared - if 5 consecutive CI tests pass I will consider this solid.

Ok, all 5 runs after a lint fix are totally green.

clemire commented:
Note: I can drop the "migrated" column in a follow-up PR, I did not do that here.
